packages/snaps-controllers/src/snaps/endowments/name-lookup.ts

```diff
@@ -94,21 +112,31 @@ function validateCaveat(caveat: Caveat<string, any>): void {
 export function getNameLookupCaveatMapper(
   value: Json,
 ): Pick<PermissionConstraint, 'caveats'> {
-  if (
-    !value ||
-    !Array.isArray(value) ||
-    (Array.isArray(value) && value.length === 0)
-  ) {
+  if (!value || !isObject(value) || Object.keys(value).length === 0) {
     return { caveats: null };
   }
-  return {
-    caveats: [
-      {
-        type: SnapCaveatType.ChainIds,
-        value,
-      },
-    ],
-  };
+
+  const caveats = [];
+
+  if (value.chains) {
+    caveats.push({
+      type: SnapCaveatType.ChainIds,
+      value: value.chains,
+    });
+  }
+
+  if (value.matchers) {
+    caveats.push({
+      type: SnapCaveatType.LookupMatchers,
+      value: value.matchers,
+    });
+  }
+
+  if (caveats.length === 0) {
+    return { caveats: null };
```

Comment:
Please test!

Wait, is this even a possibility?

Reply:
Yeah, both caveats are optional.

Reply:
The first condition can be passed with a non-empty object. We don't do full caveat validation until we hit the `PermissionController`, so this is a step before.

Reply:
If you pass a non-empty object that doesn't conform to our standards should it not fail in the PermissionController? If you do this, it will be accepted as valid 🤔

Reply:
oh that's a good point, I'll change.

Reply:
but regardless...this caveat mapper is expected to return an array of caveat objects with type etc...I wouldn't know how to fit other properties on the value object?

Reply:
I suppose I can throw instead of returning `null`?